spaceship/lib/spaceship/portal/portal_client.rb

```diff
@@ -714,7 +726,7 @@ def revoke_key!(id: nil)
 
     # This is a cache of entity type (App, AppGroup, Certificate, Device) to csrf_tokens
     def csrf_cache
-      @csrf_cache || {}
```

Comment:
Looks like a total blunder in the existing code!

Reply:
Yes, they added the caching and never validated that it actually did anything.